storage/redis.go

```diff
@@ -334,8 +334,13 @@ func (r *RedisClusterStorageManager) GetAndDeleteSet(keyName string) []interface
 
 	var lrange *redis.StringSliceCmd
 	_, err := r.db.TxPipelined(func(pipe redis.Pipeliner) error {
-		lrange = pipe.LRange(fixedKey, 0, -1)
-		pipe.Del(fixedKey)
+		lrange = pipe.LRange(fixedKey, 0, chunkSize-1)
+
+		if chunkSize == 0 {
+			pipe.Del(fixedKey)
+		} else {
+			pipe.LTrim(fixedKey, chunkSize, -1)
```

Comment:
@asoorm I just realized it won't work with gateway storage expiration mechanism. 
When we doing "Del" we remove the key, and its expiration. 
If we do just LTrim, there is a chance that analytics record will expire with active data.

Reply:
In which case, it should be the pump that is responsible for extending TTL to storage_expiration_time after a successful LTRIM and not the gateway.

Reply:
@asoorm @buger I've been playing with this and it seems that pump expiration > gateway expiration. 
I'm running this with a low `purge_chunk`, default `storage_expiration_time` (60s), and constants requests against an API and it seems that the analytics key in the Redis keeps updating the TTL to the `storage_expiration_time` so I don't think we have a problem with this.